api/tests/devcomp/unit/infrastructure/datasources/learning-content/validation/module-validation_test.js

```diff
@@ -14,59 +16,284 @@ import {
+              },
+            ],
+          },
+        ],
+      };
+
+      try {
+        await moduleSchema.validateAsync(invalidModule, { abortEarly: false });
+        throw new Error('Joi validation should have thrown');
+      } catch (joiError) {
+        expect(joiError.message).to.deep.equal(
+          '"title" failed custom validation because HTML is not allowed in this field',
+        );
+      }
+    });
+
+    it('should throw htmlNotAllowedSchema custom error for details.description field', async function () {
+      // given
+      const invalidModuleDetails = {
+        image: 'https://images.pix.fr/modulix/placeholder-details.svg',
+        description: '<strong>Découvrez avec ce didacticiel</strong> comment fonctionne Modulix !',
+        duration: 5,
+        level: 'Débutant',
+        objectives: ['Naviguer dans Modulix', 'Découvrir les leçons et les activités'],
+      };
+
+      try {
+        await moduleDetailsSchema.validateAsync(invalidModuleDetails, { abortEarly: false });
+        throw new Error('Joi validation should have thrown');
+      } catch (joiError) {
+        expect(joiError.message).to.deep.equal(
+          '"description" failed custom validation because HTML is not allowed in this field',
+        );
+      }
+    });
+
+    it('should throw htmlNotAllowedSchema custom error for details.objectives fields', async function () {
+      // given
+      const invalidModuleDetails = {
+        image: 'https://images.pix.fr/modulix/placeholder-details.svg',
+        description: 'Découvrez avec ce didacticiel comment fonctionne Modulix !',
+        duration: 5,
+        level: 'Débutant',
+        objectives: ['<span>Naviguer dans Modulix<span>', 'Découvrir les leçons et les activités'],
+      };
+
+      try {
+        await moduleDetailsSchema.validateAsync(invalidModuleDetails, { abortEarly: false });
+        throw new Error('Joi validation should have thrown');
+      } catch (joiError) {
+        expect(joiError.message).to.deep.equal(
+          '"objectives[0]" failed custom validation because HTML is not allowed in this field',
+        );
+      }
+    });
+
+    it('should throw htmlNotAllowedSchema custom error for grains.title field', async function () {
+      // given
+      const invalidModuleGrains = {
+        id: '6282925d-4775-4bca-b513-4c3009ec5886',
+        slug: 'didacticiel-modulix',
+        title: 'Didacticiel Modulix',
+        details: {
+          image: 'https://images.pix.fr/modulix/placeholder-details.svg',
+          description: 'Découvrez avec ce didacticiel comment fonctionne Modulix !',
+          duration: 5,
+          level: 'Débutant',
+          objectives: ['Naviguer dans Modulix', 'Découvrir les leçons et les activités'],
+        },
+        grains: [
+          {
+            id: '34d225e8-5d52-4ebd-9acd-8bde8438cfc9',
+            type: 'lesson',
+            title: '<strong>Sûr de ton adresse mail ?</strong>',
+            elements: [],
```

Comment:
il faut au moins un élément je pense, ce doit être pour ça que le test n'a pas passé dans la CI ⏬ 

```
      + expected - actual

      -"grains[0].title" failed custom validation because HTML is not allowed in this field. "grains" does not contain 1 required value(s)
      +"grains[0].title" failed custom validation because HTML is not allowed in this field
```

Reply:
Même avec un élément ou plus le test casse avec la même erreur.
C’est super étrange parce que lorsqu’on enlève la balise il casse pas 😫